databases/week1/database_week1.sql

```diff
@@ -0,0 +1,65 @@
+CREATE database_week1
+    DEFAULT CHARACTER SET = "utf8mb4";
+-- Find out how many tasks are in the task table
+SELECT 
+COUNT(id) as task_amount 
+FROM 
+task;
+
+-- Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id) 
+from task 
+WHERE due_date is null;
+
+-- Find all the tasks that are marked as done
+SELECT *
+FROM task 
+JOIN status
+on task.status_id = status.id
+where name = "done";
+
+-- Find all the tasks that are not marked as done
+SELECT *
+FROM task 
+JOIN status
+ON task.status_id = status.id
+WHERE name != "done";
+
+-- Get all the tasks, sorted with the most recently created first
+SELECT * 
+FROM task
+ORDER BY due_date DESC;
+
+-- Get the single most recently created task
+SELECT MAX(created)
+FROM task;
+
+-- Get the title and due date of all tasks,
+-- where the title or description contains database
+SELECT title, due_date
+FROM task
+WHERE title LIKE "%database%" OR
+description LIKE "%database%";
+
+-- Get the title and status (as text) of all tasks
+SELECT title, name
+FROM task
+JOIN status
+ON status.id = task.status_id;
+
+-- Get the name of each status,
+-- along with a count of how many tasks have that status
+SELECT name, COUNT(task.id) as Tasks
+FROM status
+JOIN task
+ON status.id = task.status_id
+GROUP BY status.name;
+
+-- Get the names of all statuses,
+-- sorted by the status with most tasks first
+SELECT name, COUNT(task.id) as Tasks
+FROM status
+JOIN task
+ON status.id = task.status_id
+GROUP BY status.name
+ORDER BY status.name ASC;
```

Comment:
This is very close as well - at the end, you order based on the status name and basically do an alphabetical ordering. However, in this exercise, you don't want to see the status on top which comes first alphabetically, but the one that has the most amount of tasks associated to it. So instead of `ORDER BY status.name ASC`, you'd have `ORDER BY Tasks DESC`.